src/OrleansRuntime/ReminderService/LocalReminderService.cs

```diff
@@ -400,7 +477,7 @@ private bool TryStopPreviousTimer(GrainReference grainRef, string reminderName)
         private async Task DoResponsibilitySanityCheck(GrainReference grainRef, string debugInfo)
         {
             if (status != ReminderServiceStatus.Started)
-                await startedTask.Task;
+                await startedTask.Task.WithTimeout(InitialLoadMaxWaitTimeForUpdates);
```

Comment:
No, it would not. 
As I wrote, you still put the `WithTimeout` (await startedTask.WithTimeout(...)), you just rethrow an explicit exc instead of a Timeout exc.